scipy/stats/_bws_test.py

```diff
@@ -0,0 +1,213 @@
+        else:
+            Bx_num = (Ri - (m + n)/n * i) * abs(Ri - (m + n)/n * i)
+            By_num = (Hj - (m + n)/n * j) * abs(Hj - (m + n)/n * j)
+
+        Bx_den = i / (n + 1) * (1 - i / (n + 1)) * m * (m + n) / n
+        By_den = j / (m + 1) * (1 - j / (m + 1)) * n * (m + n) / m
+
+        Bx = 1 / n * np.sum(Bx_num / Bx_den, axis=-1)
+        By = 1 / m * np.sum(By_num / By_den, axis=-1)
+
+        if alternative == 'greater':
+            B = (Bx - By) * 0.5
+        else:
+            B = (By - Bx) * 0.5
+
+    return B
+
+
+BWSResult = namedtuple('BWSResult', ('statistic', 'pvalue'))
+
+
+def bws_test(x, y, axis=0, alternative="two-sided", variant=None, n_resamples=9999):
+    r'''Perform the Baumgartner-Weiss-Schindler test on two independent samples.
+
+    The Baumgartner-Weiss-Schindler (BWS) test is a nonparametric test of 
+    the null hypothesis that the distribution underlying sample `x` 
+    is the same as the distribution underlying sample `y`. Unlike 
+    the Kolmogorv-Smirnov, Wilcoxon, and Cramer-Von Mises tests, 
+    the BWS test weights the integral by the variance of the difference
+    in CDFs, emphasizing the tails of the distributions, which increases
+    the power of the test for a lot of applications.
+
+    Parameters
+    ----------
+    x, y : array-like
+        N-d arrays of samples. The arrays must be broadcastable except along
+        the dimension given by `axis`.
+    axis : int, optional
+        Axis along which to perform the test. Default is 0.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis. Default is 'two-sided'.
+        Let *F(u)* and *G(u)* be the cumulative distribution functions of the
+        distributions underlying `x` and `y`, respectively. Then the following
+        alternative hypotheses are available:
+
+        * 'two-sided': the distributions are not equal, i.e. *F(u) ≠ G(u)* for
+          at least one *u*.
+        * 'less': the distribution underlying `x` is stochastically less
+          than the distribution underlying `y`, i.e. *F(u) >= G(u)* for all *u*.
+        * 'greater': the distribution underlying `x` is stochastically greater
+          than the distribution underlying `y`, i.e. *F(u) <= G(u)* for all *u*.
+
+        Under a more restrictive set of assumptions, the alternative hypotheses
+        can be expressed in terms of the locations of the distributions;
+        see [3] section 5.1.
+    variant : {'shan', 'murakami', None}, optional
+        Selects the method used to calculate the :math:`B` statistic.
+        Default is None. The following options are available.
+
+        * ``None``: use the naive method in [1]_ to calculate the :math:`B` statistic.
+        * ``'murakami'``: use the variant method in p.5 (two-sided) or p.7 (one-sided) in
+          [2]_ to calculate the :math:`B` statistic.
+        * ``'shan'``: same as ``'murakami'``.
+    n_resamples : int or np.inf, default: 9999
```

Comment:
A way to control the random state is still needed. Because of that, I'd suggest that we wait on the resolution of gh-18067 before merging this. (I recently reverted my own PR - see gh-18321 - for the same reason.)